openprescribing/matrixstore/build/precalculate_totals.py

```diff
@@ -0,0 +1,73 @@
+"""
+Pre-calculate totals for prescribing over all presentations. We sometimes need
+these values (e.g. to show prescribing of X as a percentage of all prescribing)
+and they're slightly too expensive to calculate at runtime (45-60 seconds).
+
+The resulting matrices are the same shape as the rest of the matrices and thus
+contain individual totals for each practice and month.
+"""
+import logging
+import os.path
+import sqlite3
+
+from matrixstore.open import MatrixStore
+from matrixstore.matrix_ops import is_integer, convert_to_smallest_int_type
+from matrixstore.serializer import serialize_compressed
+
+
+logger = logging.getLogger(__name__)
+
+
+def precalculate_totals(sqlite_path):
+    if not os.path.exists(sqlite_path):
+        raise RuntimeError('No SQLite file at: {}'.format(sqlite_path))
+    connection = sqlite3.connect(sqlite_path)
+    # Disable the sqlite module's magical transaction handling features because
+    # we want to use our own transactions below
+    connection.isolation_level = None
```

Comment:
For hygiene, should we reset this afterwards?

Reply:
We close the connection immediately afterwards so it seems a bit redundant to me. But we can do if you think it make things clearer.

Reply:
Agree it's redundant now, but if we add code after the commit and before closing the connection, it's possible (although unlikely) we'd miss that the isolation level wasn't what we expected.